### PR TITLE
Add prompt template retrieval endpoint

### PR DESCRIPTION
**Pull request by abubakarsohail**
This pull request adds a new endpoint for retrieving prompt templates by identifier. The endpoint is `/prompt-templates/{identifier}` and can be accessed with a POST request. This allows users to fetch prompt templates using the identifier as a parameter. The implementation includes updates to the Python SDK and JavaScript SDK to demonstrate how to use the new endpoint.

### CODE REVIEW

**Comment by jzone3** [comment]
CleanShot 2024-01-15 at 15 07 58@2x
CleanShot 2024-01-15 at 15 06 53@2x
CleanShot 2024-01-15 at 15 06 36@2x
CleanShot 2024-01-15 at 15 05 34@2x
CleanShot 2024-01-15 at 15 05 15@2x

**Review by jzone3 — CHANGES_REQUESTED** [review]
See screenshots

**Comment by abubakarsohail** [comment]
> See screenshots

@jzone3 updated the listing. the `Get Prompt Templates` is the endpoint for fetching all prompt templates.

Updated the wording for `Note`

Removed legacy code snippets.

File: reference/get-prompt-template.mdx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Get Prompt Template"
+title: "Get Prompt Template (Depcrecated)"
```
Comment (jzone3) [inline reference/get-prompt-template.mdx:2]: spelling mistake

File: features/prompt-registry.mdx [hunk 2]
```diff
@@ -64,14 +54,21 @@ Release labels like `prod` and `staging` can be optionally applied to template v
 <CodeGroup>
 
 ```python Python SDK
-promptlayer.prompts.get("my_template", label="prod")
+promptlayer.templates.get(
+  "my_template",
+  {
+    "label": "prod"
+  }
+)
```
Comment (jzone3) [inline features/prompt-registry.mdx:62]: Please fix indentation, this should be one line

File: features/prompt-registry.mdx [hunk 3]
```diff
@@ -64,14 +54,21 @@ Release labels like `prod` and `staging` can be optionally applied to template v
...
 ```
 
 ```js JavaScript
-const template_dict = await promptlayer.prompts.get({
-  prompt_name: "my_template",
-  label: "prod"
-});
+const template_dict = await promptlayer.templates.get(
```
Comment (jzone3) [inline features/prompt-registry.mdx:66]: One line

File: features/prompt-registry.mdx [hunk 4]
```diff
@@ -83,13 +80,22 @@ You can also optionally pass `version` to get an older version of a prompt. By d
 <CodeGroup>
 
 ```python Python SDK
-template_dict = promptlayer.prompts.get("my_template", version=3)
+template_dict = promptlayer.templates.get(
```
Comment (jzone3) [inline features/prompt-registry.mdx:83]: One line

File: features/prompt-registry.mdx [hunk 5]
```diff
@@ -83,13 +80,22 @@ You can also optionally pass `version` to get an older version of a prompt. By d
...
+    "version": 3
+  }
+)
 ```
 
 ```js JavaScript
-const template_dict = await promptlayer.prompts.get({
-  prompt_name: "my_template", version: 3
-});
+const template_dict = await promptlayer.templates.get(
```
Comment (jzone3) [inline features/prompt-registry.mdx:92]: one line

File: features/prompt-registry.mdx [hunk 6]
```diff
@@ -32,30 +32,20 @@ It’s simple.
...
-```js JavaScript (Legacy)
-const template_dict = await promptlayer.prompts.get({ prompt_name: "my_template" });
-```
-
 </CodeGroup>
 
 Alternatively, use the REST API endpoint `/rest/get-prompt-template` ([read more](/reference/get-prompt-template)).
 
-_Note:_ We have a new prompt template schema that is more flexible and powerful. If you are using the new schema, you can use the `promptlayer.templates.get` method to get a template. If you are using the old schema, you can use the `promptlayer.prompts.get` method to get a template. Notice how `promptlayer.prompts.get` will always return templates in legacy shape, while `promptlayer.templates.get` will always return templates in the new shape.
+*Note:* We have deprecated the legacy prompt template schema. Use `promptlayer.templates.get` to retrieve prompt templates. Notice how `promptlayer.prompts.get` will always return templates in legacy shape.
```
Comment (jzone3) [inline features/prompt-registry.mdx:46]: Note: Please use `promptlayer.templates.get` to access prompt templates. The `promptlayer.prompts.get` function is deprecated and returns templates in the legacy format.

File: features/prompt-registry.mdx [hunk 7]
```diff
@@ -32,30 +32,20 @@ It’s simple.
...
-
 </CodeGroup>
 
 Alternatively, use the REST API endpoint `/rest/get-prompt-template` ([read more](/reference/get-prompt-template)).
 
-_Note:_ We have a new prompt template schema that is more flexible and powerful. If you are using the new schema, you can use the `promptlayer.templates.get` method to get a template. If you are using the old schema, you can use the `promptlayer.prompts.get` method to get a template. Notice how `promptlayer.prompts.get` will always return templates in legacy shape, while `promptlayer.templates.get` will always return templates in the new shape.
+*Note:* We have deprecated the legacy prompt template schema. Use `promptlayer.templates.get` to retrieve prompt templates. Notice how `promptlayer.prompts.get` will always return templates in legacy shape. 
 
-For REST API of fetching a prompt template in new schema, please refer to `/prompt-templates` ([read more](/reference/templates-get)).
+For REST API of fetching a prompt template, please refer to `/prompt-templates` ([read more](/reference/templates-get)).
```
Comment (jzone3) [inline features/prompt-registry.mdx:48]: To fetch a prompt template using the REST API, refer to the endpoint endpoint `/prompt-templates` ([read more](/reference/templates-get)).

**Review by jzone3 — CHANGES_REQUESTED** [review]
A few comments. Doesn't look like you updated the REST API list that I asked for in the screenshot

File: features/prompt-registry.mdx (hunk 5 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx:92]: One line? Do you wane me to move print statement to one line?

File: features/prompt-registry.mdx [hunk 8]
```diff
@@ -487,7 +487,38 @@ The following code snippet shows you how to do this:
...
+  "my_prompt_with_functions",
+  {
+    "provider": "openai",
+    "input_variables": {
+      "location": "Boston, MA",
+      "unit": "fahrenheit",
+    }
+  }
+)
+response = openai.chat.completions.create(
```
Comment (jzone3) [inline features/prompt-registry.mdx:515]: Why is this `openai.chat` instead of `client.chat`?

**Review by jzone3 — CHANGES_REQUESTED** [review]

File: features/prompt-registry.mdx (hunk 8 above)
Comment (abubakarsohail) [reply features/prompt-registry.mdx:515]: updated to client

**Pull request closed without merging**